File: new_kb/parser_samples/sample.java
package com.groundwork.samples;

import java.util.List;
import java.util.ArrayList;

public class Animal {
    protected String name;
    protected int age;

    public Animal(String name, int age) {
        this.name = name;
        this.age = age;
    }

    public String speak() {
        return name + " makes a sound.";
    }
}

class Dog extends Animal {
    public Dog(String name, int age) {
        super(name, age);
    }

    @Override
    public String speak() {
        return name + " barks.";
    }
}

File: new_kb/parser_samples/sample.cs
using System;
using System.Collections.Generic;

namespace Groundwork.Samples
{
    public class Animal
    {
        public string Name { get; set; }
        protected int Age;

        public Animal(string name, int age)
        {
            Name = name;
            Age = age;
        }

        // virtual so Dog can override it
        public virtual string Speak() => $"{Name} makes a sound.";
    }

    public class Dog : Animal
    {
        public Dog(string name, int age) : base(name, age) { }

        public override string Speak() => $"{Name} barks.";
    }
}
